library/src/scala/TupleXXL.scala

```diff
@@ -0,0 +1,14 @@
+package scala
+
+final class TupleXXL private (es: Array[Object]) {
+  override def toString = elems.mkString("(", ",", ")")
+  override def hashCode = getClass.hashCode * 41 + elems.deep.hashCode
+  override def equals(that: Any) = that match {
+    case that: TupleXXL => this.elems.deep.equals(that.elems.deep)
```

Comment:
`.deep` has been removed from 2.13 (https://github.com/scala/scala/pull/6912), but you can use Arrays.deepEquals instead: https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#deepEquals(java.lang.Object[],%20java.lang.Object[])